internal/pkg/service/common/etcdop/watchmirror_test.go

```diff
@@ -0,0 +1,104 @@
+package etcdop
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	clientv3 "go.etcd.io/etcd/client/v3"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/serde"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/etcdhelper"
+)
+
+type testUser struct {
+	FirstName string
+	LastName  string
+	Age       int
+}
+
+func TestMirror(t *testing.T) {
+	t.Parallel()
+
+	wg := &sync.WaitGroup{}
+	defer wg.Wait()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+	defer cancel()
+
+	// Create a typed prefix with some keys
+	client := etcdhelper.ClientForTest(t, etcdhelper.TmpNamespace(t))
+	pfx := NewTypedPrefix[testUser]("my/prefix", serde.NewJSON(serde.NoValidation))
+	require.NoError(t, pfx.Key("key1").Put(testUser{FirstName: "John", LastName: "Brown", Age: 10}).DoOrErr(ctx, client))
+	require.NoError(t, pfx.Key("key2").Put(testUser{FirstName: "Paul", LastName: "Green", Age: 20}).DoOrErr(ctx, client))
+
+	// Setup mirroring of a prefix tree to the memory, with custom key and value mapping.
+	// The result are in-memory KV pairs "<first name> <last name>" => <age>.
+	logger := log.NewDebugLogger()
+	mirror, errCh := SetupMirror(
+		logger,
+		pfx.GetAllAndWatch(ctx, client, clientv3.WithPrevKV()),
+		func(kv *op.KeyValue, v testUser) string { return v.FirstName + " " + v.LastName },
+		func(kv *op.KeyValue, v testUser) int { return v.Age },
```

Comment:
Pomocou tychto callbacks sa mapuje key a value. 
Typ `V` je v tomto pripade `int`, .... podla toho aky tu dame typ, tak vsetky ostatne/genericke metody pracuju s tymto typom. Moze to byt aj nieco zlozitejsie, struktura, slice, ....